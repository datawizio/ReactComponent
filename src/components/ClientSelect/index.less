@import "../../styles/variables";
@import "./rtl";

.ant-select.client-select.ant-select-show-search.ant-select-single {
  width: 120px;
  .ant-select-selector {
    cursor: pointer;
    border: 0;
    padding-right: 0;
  }
  .ant-select-selection-item {
    text-align: right;
    padding-right: 19px;
  }
  .ant-select-selection-search-input {
    cursor: pointer;
  }
  .ant-select-arrow {
    color: rgba(0, 0, 0, 0.65);
    right: 0;
    cursor: pointer;
  }
  .ant-select-selection-search {
    right: 16px;
  }
  &.ant-select-open {
    .ant-select-selection-search-input {
      cursor: text;
    }
  }
}

.divider {
  height: 100%;
  border-right: 1px solid #e9e9e9;
  margin: 0 @margin-md;
}
