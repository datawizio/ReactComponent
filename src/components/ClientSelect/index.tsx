import React from "react";

import Select from "../Select";

import { SelectValue } from "antd/lib/select";

import "./index.less";

export interface ClientSelectProps {
  clients: [{ id: number; name: string }];
  client: number;
}

const ClientSelect: React.FC<ClientSelectProps> = ({ clients, client }) => {
  const handleChange = (value: SelectValue) => {
    const arr = window.location.pathname.split("/").slice(3);
    window.location.href = `/c/${value}/${arr.join("/")}`;
  };

  return (
    <>
      {clients.length === 1 ? (
        clients[0].name
      ) : (
        <Select
          value={client}
          showSearch
          optionFilterProp="label"
          onChange={handleChange}
          className="client-select"
        >
          {clients.map(client => (
            <Select.Option key={client.id} value={client.id}>
              {client.name}
            </Select.Option>
          ))}
        </Select>
      )}
      <div className="divider"></div>
    </>
  );
};

export default ClientSelect;
