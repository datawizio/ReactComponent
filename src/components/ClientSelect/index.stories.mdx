import { Meta, Story, Preview, Props } from "@storybook/addon-docs/blocks";
import ClientSelect from "./index";

<Meta title="Components/Select/ClientSelect" component={ClientSelect} />

# ClientSelect

Компонент

## Basic

<Preview>
  <Story name="Basic">
    <ClientSelect
      clients={[
        { name: "Ololo", id: 1 },
        { name: "Ololo2", id: 2 }
      ]}
      client={1}
    />
  </Story>
</Preview>

## Props

<Props of={ClientSelect} />
