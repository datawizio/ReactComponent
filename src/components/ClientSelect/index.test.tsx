import "jsdom-global/register";
import React from "react";
import { mount } from "enzyme";

import ClientSelect from "./index";
const clients = [
  {
    id: 1,
    name: "cl1"
  },
  {
    id: 2,
    name: "cl2"
  }
];
const mockProps = {
  clients,
  client: 1,
  onChange: jest.fn()
};

const setUp = (props?) => mount(<ClientSelect {...props} />);

describe("ClientSelect component", () => {
  let component;
  beforeEach(() => {
    component = setUp(mockProps);
  });

  it("ClientSelect rendered correctly", () => {
    expect(component).toMatchSnapshot();
  });

  it("ClientSelect with one child", () => {
    const wrapper = setUp({ ...mockProps, clients: [clients[0]] });
    expect(wrapper.html()).toContain(clients[0].name);
  });

  it("ClientSelect onChange", () => {
    Object.defineProperty(window, "location", {
      value: {
        pathname: "/c/willReplace/report/qwerty"
      }
    });
    const mockValue = "value1";
    const select = component.find(".client-select").first();
    const onChange = component.props().onChange;
    select.prop("onChange")(mockValue);
    expect(onChange).toBeCalled();
    expect(onChange).toBeCalledWith(`/c/${mockValue}/report/qwerty`);
  });
});
