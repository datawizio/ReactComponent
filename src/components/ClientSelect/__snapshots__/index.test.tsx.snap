// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientSelect component ClientSelect rendered correctly 1`] = `
<ClientSelect
  client={1}
  clients={
    Array [
      Object {
        "id": 1,
        "name": "cl1",
      },
      Object {
        "id": 2,
        "name": "cl2",
      },
    ]
  }
  onChange={[MockFunction]}
>
  <Select
    asyncData={false}
    className="client-select"
    loadingContent="Loading..."
    notFoundContent="No data"
    onChange={[Function]}
    optionFilterProp="label"
    showSearch={true}
    value={1}
    withPagination={false}
  >
    <ForwardRef(InternalSelect)
      className="client-select"
      loading={false}
      notFoundContent="No data"
      onChange={[Function]}
      optionFilterProp="label"
      optionLabelProp="label"
      showSearch={true}
      tagRender={null}
      value={1}
    >
      <Select
        className="client-select"
        clearIcon={<ForwardRef(CloseCircleFilled) />}
        dropdownClassName=""
        inputIcon={[Function]}
        listHeight={256}
        listItemHeight={24}
        loading={false}
        menuItemSelectedIcon={null}
        notFoundContent="No data"
        onChange={[Function]}
        optionFilterProp="label"
        optionLabelProp="label"
        prefixCls="ant-select"
        removeIcon={<ForwardRef(CloseOutlined) />}
        showSearch={true}
        tagRender={null}
        transitionName="slide-up"
        value={1}
      >
        <ForwardRef(Select)
          className="client-select"
          clearIcon={<ForwardRef(CloseCircleFilled) />}
          dropdownClassName=""
          inputIcon={[Function]}
          listHeight={256}
          listItemHeight={24}
          loading={false}
          menuItemSelectedIcon={null}
          notFoundContent="No data"
          onChange={[Function]}
          optionFilterProp="label"
          optionLabelProp="label"
          prefixCls="ant-select"
          removeIcon={<ForwardRef(CloseOutlined) />}
          showSearch={true}
          tagRender={null}
          transitionName="slide-up"
          value={1}
        >
          <div
            className="ant-select client-select ant-select-single ant-select-show-arrow ant-select-show-search"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
          >
            <SelectTrigger
              containerWidth={null}
              dropdownClassName=""
              empty={false}
              getTriggerDOMNode={[Function]}
              popupElement={
                <ForwardRef(OptionList)
                  childrenAsData={true}
                  defaultActiveFirstOption={true}
                  flattenOptions={
                    Array [
                      Object {
                        "data": Object {
                          "children": "cl1",
                          "key": "1",
                          "label": "cl1",
                          "value": 1,
                        },
                        "groupOption": false,
                        "key": "1",
                      },
                      Object {
                        "data": Object {
                          "children": "cl2",
                          "key": "2",
                          "label": "cl2",
                          "value": 2,
                        },
                        "groupOption": false,
                        "key": "2",
                      },
                    ]
                  }
                  height={256}
                  id="rc_select_TEST_OR_SSR"
                  itemHeight={24}
                  menuItemSelectedIcon={null}
                  multiple={false}
                  notFoundContent="No data"
                  onActiveValue={[Function]}
                  onMouseEnter={[Function]}
                  onSelect={[Function]}
                  onToggleOpen={[Function]}
                  options={
                    Array [
                      Object {
                        "children": "cl1",
                        "key": "1",
                        "label": "cl1",
                        "value": 1,
                      },
                      Object {
                        "children": "cl2",
                        "key": "2",
                        "label": "cl2",
                        "value": 2,
                      },
                    ]
                  }
                  prefixCls="ant-select"
                  searchValue=""
                  values={
                    Set {
                      1,
                    }
                  }
                  virtual={true}
                />
              }
              prefixCls="ant-select"
              transitionName="slide-up"
            >
              <Trigger
                action={Array []}
                afterPopupVisibleChange={[Function]}
                autoDestroy={false}
                blurDelay={0.15}
                builtinPlacements={
                  Object {
                    "bottomLeft": Object {
                      "offset": Array [
                        0,
                        4,
                      ],
                      "overflow": Object {
                        "adjustX": 0,
                        "adjustY": 1,
                      },
                      "points": Array [
                        "tl",
                        "bl",
                      ],
                    },
                    "bottomRight": Object {
                      "offset": Array [
                        0,
                        4,
                      ],
                      "overflow": Object {
                        "adjustX": 0,
                        "adjustY": 1,
                      },
                      "points": Array [
                        "tr",
                        "br",
                      ],
                    },
                    "topLeft": Object {
                      "offset": Array [
                        0,
                        -4,
                      ],
                      "overflow": Object {
                        "adjustX": 0,
                        "adjustY": 1,
                      },
                      "points": Array [
                        "bl",
                        "tl",
                      ],
                    },
                    "topRight": Object {
                      "offset": Array [
                        0,
                        -4,
                      ],
                      "overflow": Object {
                        "adjustX": 0,
                        "adjustY": 1,
                      },
                      "points": Array [
                        "br",
                        "tr",
                      ],
                    },
                  }
                }
                defaultPopupVisible={false}
                destroyPopupOnHide={false}
                focusDelay={0}
                getDocument={[Function]}
                getPopupClassNameFromAlign={[Function]}
                getTriggerDOMNode={[Function]}
                hideAction={Array []}
                mask={false}
                maskClosable={true}
                mouseEnterDelay={0}
                mouseLeaveDelay={0.1}
                onPopupAlign={[Function]}
                onPopupVisibleChange={[Function]}
                popup={
                  <div>
                    <ForwardRef(OptionList)
                      childrenAsData={true}
                      defaultActiveFirstOption={true}
                      flattenOptions={
                        Array [
                          Object {
                            "data": Object {
                              "children": "cl1",
                              "key": "1",
                              "label": "cl1",
                              "value": 1,
                            },
                            "groupOption": false,
                            "key": "1",
                          },
                          Object {
                            "data": Object {
                              "children": "cl2",
                              "key": "2",
                              "label": "cl2",
                              "value": 2,
                            },
                            "groupOption": false,
                            "key": "2",
                          },
                        ]
                      }
                      height={256}
                      id="rc_select_TEST_OR_SSR"
                      itemHeight={24}
                      menuItemSelectedIcon={null}
                      multiple={false}
                      notFoundContent="No data"
                      onActiveValue={[Function]}
                      onMouseEnter={[Function]}
                      onSelect={[Function]}
                      onToggleOpen={[Function]}
                      options={
                        Array [
                          Object {
                            "children": "cl1",
                            "key": "1",
                            "label": "cl1",
                            "value": 1,
                          },
                          Object {
                            "children": "cl2",
                            "key": "2",
                            "label": "cl2",
                            "value": 2,
                          },
                        ]
                      }
                      prefixCls="ant-select"
                      searchValue=""
                      values={
                        Set {
                          1,
                        }
                      }
                      virtual={true}
                    />
                  </div>
                }
                popupAlign={Object {}}
                popupClassName=""
                popupPlacement="bottomLeft"
                popupStyle={
                  Object {
                    "minWidth": null,
                    "width": null,
                  }
                }
                popupTransitionName="slide-up"
                prefixCls="ant-select-dropdown"
                showAction={Array []}
              >
                <Selector
                  accessibilityIndex={0}
                  activeValue={null}
                  className="client-select"
                  clearIcon={<ForwardRef(CloseCircleFilled) />}
                  domRef={
                    Object {
                      "current": <div
                        class="ant-select-selector"
                      >
                        <span
                          class="ant-select-selection-search"
                        >
                          <input
                            aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                            aria-autocomplete="list"
                            aria-controls="rc_select_TEST_OR_SSR_list"
                            aria-haspopup="listbox"
                            aria-owns="rc_select_TEST_OR_SSR_list"
                            autocomplete="off"
                            class="ant-select-selection-search-input"
                            id="rc_select_TEST_OR_SSR"
                            role="combobox"
                            type="search"
                            value=""
                          />
                        </span>
                        <span
                          class="ant-select-selection-item"
                          title="cl1"
                        >
                          cl1
                        </span>
                      </div>,
                    }
                  }
                  dropdownClassName=""
                  id="rc_select_TEST_OR_SSR"
                  inputElement={null}
                  inputIcon={[Function]}
                  key="trigger"
                  listHeight={256}
                  listItemHeight={24}
                  loading={false}
                  menuItemSelectedIcon={null}
                  multiple={false}
                  notFoundContent="No data"
                  onChange={[Function]}
                  onSearch={[Function]}
                  onSearchSubmit={[Function]}
                  onSelect={[Function]}
                  onToggleOpen={[Function]}
                  optionFilterProp="label"
                  optionLabelProp="label"
                  prefixCls="ant-select"
                  removeIcon={<ForwardRef(CloseOutlined) />}
                  searchValue=""
                  showSearch={true}
                  tagRender={null}
                  tokenWithEnter={false}
                  transitionName="slide-up"
                  value={1}
                  values={
                    Array [
                      Object {
                        "disabled": undefined,
                        "key": 1,
                        "label": "cl1",
                        "value": 1,
                      },
                    ]
                  }
                >
                  <div
                    className="ant-select-selector"
                    onClick={[Function]}
                    onMouseDown={[Function]}
                  >
                    <SingleSelector
                      accessibilityIndex={0}
                      activeValue={null}
                      className="client-select"
                      clearIcon={<ForwardRef(CloseCircleFilled) />}
                      domRef={
                        Object {
                          "current": <div
                            class="ant-select-selector"
                          >
                            <span
                              class="ant-select-selection-search"
                            >
                              <input
                                aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                                aria-autocomplete="list"
                                aria-controls="rc_select_TEST_OR_SSR_list"
                                aria-haspopup="listbox"
                                aria-owns="rc_select_TEST_OR_SSR_list"
                                autocomplete="off"
                                class="ant-select-selection-search-input"
                                id="rc_select_TEST_OR_SSR"
                                role="combobox"
                                type="search"
                                value=""
                              />
                            </span>
                            <span
                              class="ant-select-selection-item"
                              title="cl1"
                            >
                              cl1
                            </span>
                          </div>,
                        }
                      }
                      dropdownClassName=""
                      id="rc_select_TEST_OR_SSR"
                      inputElement={null}
                      inputIcon={[Function]}
                      inputRef={
                        Object {
                          "current": <input
                            aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                            aria-autocomplete="list"
                            aria-controls="rc_select_TEST_OR_SSR_list"
                            aria-haspopup="listbox"
                            aria-owns="rc_select_TEST_OR_SSR_list"
                            autocomplete="off"
                            class="ant-select-selection-search-input"
                            id="rc_select_TEST_OR_SSR"
                            role="combobox"
                            type="search"
                            value=""
                          />,
                        }
                      }
                      listHeight={256}
                      listItemHeight={24}
                      loading={false}
                      menuItemSelectedIcon={null}
                      multiple={false}
                      notFoundContent="No data"
                      onChange={[Function]}
                      onInputChange={[Function]}
                      onInputCompositionEnd={[Function]}
                      onInputCompositionStart={[Function]}
                      onInputKeyDown={[Function]}
                      onInputMouseDown={[Function]}
                      onInputPaste={[Function]}
                      onSearch={[Function]}
                      onSearchSubmit={[Function]}
                      onSelect={[Function]}
                      onToggleOpen={[Function]}
                      optionFilterProp="label"
                      optionLabelProp="label"
                      prefixCls="ant-select"
                      removeIcon={<ForwardRef(CloseOutlined) />}
                      searchValue=""
                      showSearch={true}
                      tagRender={null}
                      tokenWithEnter={false}
                      transitionName="slide-up"
                      value={1}
                      values={
                        Array [
                          Object {
                            "disabled": undefined,
                            "key": 1,
                            "label": "cl1",
                            "value": 1,
                          },
                        ]
                      }
                    >
                      <span
                        className="ant-select-selection-search"
                      >
                        <Input
                          accessibilityIndex={0}
                          attrs={Object {}}
                          editable={true}
                          id="rc_select_TEST_OR_SSR"
                          inputElement={null}
                          onChange={[Function]}
                          onCompositionEnd={[Function]}
                          onCompositionStart={[Function]}
                          onKeyDown={[Function]}
                          onMouseDown={[Function]}
                          onPaste={[Function]}
                          prefixCls="ant-select"
                          value=""
                        >
                          <input
                            aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                            aria-autocomplete="list"
                            aria-controls="rc_select_TEST_OR_SSR_list"
                            aria-haspopup="listbox"
                            aria-owns="rc_select_TEST_OR_SSR_list"
                            autoComplete="off"
                            className="ant-select-selection-search-input"
                            id="rc_select_TEST_OR_SSR"
                            onChange={[Function]}
                            onCompositionEnd={[Function]}
                            onCompositionStart={[Function]}
                            onKeyDown={[Function]}
                            onMouseDown={[Function]}
                            onPaste={[Function]}
                            readOnly={false}
                            role="combobox"
                            style={
                              Object {
                                "opacity": null,
                              }
                            }
                            type="search"
                            unselectable={null}
                            value=""
                          />
                        </Input>
                      </span>
                      <span
                        className="ant-select-selection-item"
                        title="cl1"
                      >
                        cl1
                      </span>
                    </SingleSelector>
                  </div>
                </Selector>
              </Trigger>
            </SelectTrigger>
            <TransBtn
              className="ant-select-arrow"
              customizeIcon={[Function]}
              customizeIconProps={
                Object {
                  "focused": false,
                  "loading": false,
                  "open": undefined,
                  "searchValue": "",
                  "showSearch": true,
                }
              }
            >
              <span
                aria-hidden={true}
                className="ant-select-arrow"
                onMouseDown={[Function]}
                style={
                  Object {
                    "WebkitUserSelect": "none",
                    "userSelect": "none",
                  }
                }
                unselectable="on"
              >
                <ForwardRef(DownOutlined)
                  className="ant-select-suffix"
                >
                  <AntdIcon
                    className="ant-select-suffix"
                    icon={
                      Object {
                        "icon": Object {
                          "attrs": Object {
                            "focusable": "false",
                            "viewBox": "64 64 896 896",
                          },
                          "children": Array [
                            Object {
                              "attrs": Object {
                                "d": "M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z",
                              },
                              "tag": "path",
                            },
                          ],
                          "tag": "svg",
                        },
                        "name": "down",
                        "theme": "outlined",
                      }
                    }
                  >
                    <span
                      aria-label="down"
                      className="anticon anticon-down ant-select-suffix"
                      role="img"
                    >
                      <IconReact
                        icon={
                          Object {
                            "icon": Object {
                              "attrs": Object {
                                "focusable": "false",
                                "viewBox": "64 64 896 896",
                              },
                              "children": Array [
                                Object {
                                  "attrs": Object {
                                    "d": "M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z",
                                  },
                                  "tag": "path",
                                },
                              ],
                              "tag": "svg",
                            },
                            "name": "down",
                            "theme": "outlined",
                          }
                        }
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="down"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          key="svg-down"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                            key="svg-down-svg-0"
                          />
                        </svg>
                      </IconReact>
                    </span>
                  </AntdIcon>
                </ForwardRef(DownOutlined)>
              </span>
            </TransBtn>
          </div>
        </ForwardRef(Select)>
      </Select>
    </ForwardRef(InternalSelect)>
  </Select>
  <div
    className="divider"
  />
</ClientSelect>
`;
